internal/backend/auth/authhttpmiddleware/authmiddleware.go

```diff
@@ -104,3 +105,21 @@ func New(deps Deps) AuthMiddlewareDecorator {
 		return f
 	}
 }
+
+type AuthHeaderExtractor func(*http.Request) sdktypes.User
+
+// just for logs in interceptor
```

Comment:
This comment doesn't actually explain what this functions does, or why/when.

Reply:
added more comment lines